### PR TITLE
Adapt to 0.9.0

### PR DESCRIPTION
**Pull request by FabijanC**
- Updated expected addresses for salty deploys
- Fix some dependency issues (update to hardhat 2.9.9)
- Update tests to use maxFee on invoke.
- Add scripts for:
  - Argent account deployment (deploy-argent.ts)
  - Fund transfer (transfer-funds.ts):
    - This required to introduce the ERC20 token contract artifacts (added the same that's used in devnet).

### CODE REVIEW

File: test/util.ts [hunk 1]
```diff
@@ -6,3 +6,31 @@ export function expectFeeEstimationStructure(fee: any) {
...
+  
+  /**
+   * Expects address equality after adapting them.
+   * @param actual 
+   * @param expected 
+   */
+  export function expectAddressEquality(actual: string, expected: string) {
+    expect(adaptAddress(actual)).to.equal(adaptAddress(expected));
+  }
+
```
Comment (Nathan-SL) [inline test/util.ts:36]: Missing EOL here

Comment (FabijanC) [reply test/util.ts]: It was actually reporting extra whitespace; fixed it: https://github.com/Shard-Labs/starknet-hardhat-example/pull/65/commits/8788e4f3845fbc8e5a1923a7d28def79fad9c5d0

File: test/util.ts [hunk 2]
```diff
@@ -6,3 +6,31 @@ export function expectFeeEstimationStructure(fee: any) {
     expect(typeof fee.amount).to.equal("bigint");
     expect(fee.unit).to.equal("wei");
 }
+
+export function ensureEnvVar(varName: string): string {
+    if (!process.env[varName]) {
+        throw new Error(`Env var ${varName} not set or empty`);
+    }
+    return <string>process.env[varName];
```
[the remark this answers is not in the document]
Comment (Nathan-SL) [reply test/util.ts:14]: Only for the sake of readability.

[the remark this answers is not in the document]
Comment (FabijanC) [reply test/util.ts]: https://github.com/Shard-Labs/starknet-hardhat-example/pull/65/commits/5d0d93dacdc092b001df7b56d9c05f4127dd801b

**Review by Nathan-SL — APPROVED** [review]

**Pull request merged**